src/rabbit_queue_index.erl

```diff
@@ -257,10 +258,19 @@
 
 erase(Name) ->
     #qistate { dir = Dir } = blank_state(Name),
-    case rabbit_file:is_dir(Dir) of
-        true  -> rabbit_file:recursive_delete([Dir]);
-        false -> ok
-    end.
+    erase_index_dir(Dir).
+
+%% used during variable queue purge when there are no pending acks
```

Comment:
we need a _usable_ index state, since after BQ:purge/1 the index state might keep being used by future publishes and so on.